python/dgl/nn/pytorch/explain/subgraphxexplainer.py

```diff
@@ -0,0 +1,566 @@
+        r"""Get the average reward for multiple visits.
+
+        Returns
+        -------
+        float
+            Return the averaged reward for multiple visits.
+        """
+        return self.total_reward / self.num_visit if self.num_visit > 0 else 0
+
+    def action_selection_criteria(self, total_visit_count):
+        r"""Get the action selection criteria of node.
+
+        Parameters
+        ----------
+        total_visit_count: float
+            The total visiting counts for all possible actions of node.
+
+        Returns
+        -------
+        float
+            Returns the action selection criteria of node.
+        """
+        return (
+            self.coef
+            * self.immediate_reward
+            * math.sqrt(total_visit_count)
+            / (1 + self.num_visit)
+        )
+
+
+class SubgraphXExplainer(nn.Module):
+    r"""SubgraphXExplainer model from `SubgraphXExplainer: On Explainability of
+    Graph Neural Networks via Subgraph Explorations <https://arxiv.org/pdf/2102.05152.pdf>`.
+
+    It identifies the most important subgraph from the original graph that plays a
+    critical role in GNN-based graph classification.
+
+    It employs Monte Carlo tree search (MCTS) in efficiently exploring different subgraphs
+    for explanation and uses Shapley values as the measure of subgraph importance.
+
+    Parameters
+    ----------
+    model: nn.Module
+        The GNN model to explain.
+
+        * The required arguments of its forward function are `DGLGraph` graph and its
+        node features.
+        * The output of its forward function is the logits for the predicted
+          graph classes.
+    num_gnn_layers: int
+        Number of layers of GNN model. Needed to calculate the l-hop neighbouring
+        nodes of graph. Computing shapley values for big and complex graphs
+        is time and resource consuming. The shapley algorithm implemented in the paper
+        efficiently approximate shapley values by finding coalition of l-hop neighboring
+        nodes of the graph, where the GNN model has l layers.
+    coef: float
+        The hyperparameter that controls the trade-off between exploration and exploitation.
+        A higher exploration rate encourages exploration of relatively unvisited nodes.
+        So, the more a node has been visited (and the more certain is its total reward value),
+        the less it will be visited. For example, if a node has been visited many times and
+        has a high total reward value, a high coef will cause the algorithm to explore other nodes
+        instead of continuing to visit that node.
+    high2low: bool, optional
+        Pruning action either "High2low" or "Low2high" (refer to paper).
+        "High2low": Whether to expand children nodes from high degree to low degree when
+        extend the child nodes in the search tree. "Low2high" is opposite of "High2low".
+        If True, it will use "High2low" pruning action, default to True.
```

Comment:
Replace the whole description with the one below.

```suggestion
        If True, it will use "High2low" pruning action, otherwise "Low2high". Default: True
        
        * "High2low": Consider the pruning actions corresponding to the nodes with top-K degree.
```

Reply:
done.